bindsnet/network/topology.py

```diff
@@ -76,8 +76,13 @@ def update(self, **kwargs) -> None:
         """
         Compute connection's update rule.
         """
+        learning = kwargs.get('learning', True)
         reward = kwargs.get('reward', None)
-        self.update_rule.update(reward=reward)
+        if learning or 'bindsnet.learning.NoOp' in str(self.update_rule.__class__):
```

Comment:
I think we should just have:

```
if learning:
    self.update_rule.update(reward=reward)
```

and just stipulate that if learning if "off", weight decay is off as well. In that sense, `learning = False` implies that weights are frozen.

Reply:
Alright, but doesn't that make `NoOp` redundant? Why not just have `update_rule` be None then?